packages/create/src/index.ts

```diff
@@ -0,0 +1,135 @@
+import fs from 'fs-extra';
+import path from 'path';
+import meow from 'meow';
+import enquirer from 'enquirer';
+import execa from 'execa';
+import ora from 'ora';
+import c from 'chalk';
+import terminalLink from 'terminal-link';
+import { checkVersion } from './checkVersion';
+import { UserError } from './utils';
+import { fileURLToPath } from 'url';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const starterDir = path.normalize(`${__dirname}/../starter`);
+
+const cli = meow(
+  `
+Usage
+  $ create-keystone-app [directory]
+`
+);
+
+type Args = {
+  directory: string;
+};
+
+const versionInfo = () => {
+  process.stdout.write('\n');
+  console.log(`✨ You're about to generate a project using ${c.bold(
+    'Keystone 6'
+  )} packages.
+`);
+};
+
+async function normalizeArgs(): Promise<Args> {
+  let directory = cli.input[0];
+  if (!directory) {
+    ({ directory } = await enquirer.prompt({
+      type: 'input',
+      name: 'directory',
+      message:
+        'What directory should create-keystone-app generate your app into?',
+      validate: (x) => !!x,
+    }));
+    process.stdout.write('\n');
+  }
+  return {
+    directory: path.resolve(directory),
+  };
+}
+
+function pkgManagerFromUserAgent(userAgent: string | undefined) {
+  if (!userAgent) return 'npm';
+  const pkgSpec = userAgent.split(' ')[0];
+  const [name, _version] = pkgSpec.split('/');
+  return name ?? 'npm';
+}
+
+const installDeps = async (cwd: string): Promise<string> => {
+  const pkgManager = pkgManagerFromUserAgent(process.env.npm_config_user_agent);
+  const spinner = ora(
+    `Installing dependencies with ${pkgManager}. This may take a few minutes.`
+  ).start();
+  try {
+    await execa(pkgManager, ['install'], { cwd });
+    spinner.succeed(`Installed dependencies with ${pkgManager}.`);
+    return pkgManager;
+  } catch (err) {
+    spinner.fail(`Failed to install with ${pkgManager}.`);
+    throw err;
+  }
+};
```

Comment:
This function was refactored to detect the package manager instead of trying yarn and then npm

`pkgManagerFromUserAgent` is an adaptation of https://github.com/vitejs/vite/blob/2d50be2a5424e4f4c22774652ed313d2a232f8af/packages/create-vite/src/index.ts#L538-L546

Reply:
I opted for something less, the format seems to be defined at https://github.com/npm/cli/blob/9214be9ed8779493e00d193e36a930918a30be64/workspaces/config/lib/definitions/definitions.js#L2107, and is slowly becoming a standard?

See [pnpm](https://github.com/pnpm/pnpm/blob/726753f05d6512c93dd6f26787a92891ded59533/exec/plugin-commands-script-runners/src/makeEnv.ts) and [yarn](https://github.com/yarnpkg/berry/blob/35167b26b20509fcc9eaaa552f1fb1e58359c3d4/packages/yarnpkg-core/sources/scriptUtils.ts#L192).

From reading the external sources, it seems we can simply take up until the first `/`:
```typescript
const [packageManager] = process.env.npm_config_user_agent?.split('/', 1) ?? ['npm']
```